kubectl-plugin/pkg/cmd/submit/submit.go

```diff
@@ -0,0 +1,357 @@
+				return err
+			}
+			if err := options.Validate(); err != nil {
+				return err
+			}
+			return options.Run(cmd.Context(), cmdFactory)
+		},
+	}
+	cmd.Flags().StringVarP(&options.fileName, "filename", "f", options.fileName, "Path and name of the Ray Job YAML file")
+	cmd.Flags().StringVar(&options.cluster, "cluster-name", options.cluster, "Name of the cluster to submit the job to.")
+	cmd.Flags().StringVar(&options.address, "address", options.address, "Address of the ray cluster to connect to")
+	cmd.Flags().StringVar(&options.submissionID, "submission-id", options.submissionID, "ID to specify for the ray job. If not provided, one will be generated")
+	cmd.Flags().StringVar(&options.runtimeEnv, "runtime-env", options.runtimeEnv, "Path and name to the runtime env YAML file")
+	cmd.Flags().StringVar(&options.workingDir, "working-dir", options.workingDir, "Directory containing files that your job will run in")
+	cmd.Flags().Float32Var(&options.entryPointCPU, "entrypoint-num-cpus", options.entryPointCPU, "Number of CPU reserved for the for the entrypoint command")
+	cmd.Flags().Float32Var(&options.entryPointGPU, "entrypoint-num-gpus", options.entryPointGPU, "Number of GPU reserved for the for the entrypoint command")
+	cmd.Flags().IntVar(&options.entryPointMemory, "entrypoint-memory", options.entryPointMemory, "Amount of memory reserved for the entrypoint command")
+	cmd.Flags().BoolVar(&options.noWait, "no-wait", options.noWait, "If present, will not stream logs and wait for job to finish")
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *SubmitJobOptions) Complete() error {
+	if *options.configFlags.Namespace == "" {
+		*options.configFlags.Namespace = "default"
+	}
+
+	if options.workingDir != "" {
+		if options.fileName == "" {
+			return fmt.Errorf("Ray Job file -f/--filename is required when using --working-dir")
+		}
+	}
+
+	if !strings.HasPrefix(options.address, "http://") {
+		options.address = "http://" + options.address
+	}
+
+	options.runtimeEnv = filepath.Clean(options.runtimeEnv)
+	return nil
+}
+
+func (options *SubmitJobOptions) Validate() error {
+	// Overrides and binds the kube config then retrieves the merged result
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("Error retrieving raw config: %w", err)
+	}
+	if len(config.CurrentContext) == 0 {
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+
+	if len(options.workingDir) > 0 {
+		info, err := os.Stat(options.workingDir)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("Working dir path does not exist. Failed with: %w", err)
+		} else if err != nil {
+			return fmt.Errorf("Error occurred when checking working dir file: %w", err)
+		} else if !info.IsDir() {
+			return fmt.Errorf("Working dir is not a directory, please use directory: %w", err)
+		}
+
+		// RayJob yaml validation
```

Comment:
Why is RayJob YAML validation only checked when working dir is specified?

Reply:
Yup, I moved it out.